internal/providers/pluginfw/resources/qualitymonitor/resource_quality_monitor_test.go

```diff
@@ -151,3 +153,43 @@ func TestUcAccUpdateQualityMonitorPluginFramework(t *testing.T) {
 		`,
 	})
 }
+
+func TestUcAccQualityMonitorImportPluginFramework(t *testing.T) {
+	if os.Getenv("GOOGLE_CREDENTIALS") != "" {
+		t.Skipf("databricks_quality_monitor resource is not available on GCP")
+	}
+	acceptance.UnityWorkspaceLevel(t, acceptance.Step{
+		Template: commonPartQualityMonitoring + `
+
+			resource "databricks_quality_monitor_pluginframework" "testMonitorInference" {
+				table_name = databricks_sql_table.myInferenceTable.id
+				assets_dir = "/Shared/provider-test/databricks_quality_monitoring/${databricks_sql_table.myInferenceTable.name}"
+				output_schema_name = databricks_schema.things.id
+				inference_log = {
+				  granularities = ["1 day"]
+				  timestamp_col = "timestamp"
+				  prediction_col = "prediction"
+				  model_id_col = "model_id"
+				  problem_type = "PROBLEM_TYPE_REGRESSION"
+				} 
+			}
+
+			output "table_name" {
+				value = databricks_quality_monitor_pluginframework.testMonitorInference.table_name
+			}
+		`,
+	},
+		acceptance.Step{
+			ImportState:  true,
+			ResourceName: "databricks_quality_monitor_pluginframework.testMonitorInference",
+			ImportStateIdFunc: func(s *terraform.State) (string, error) {
+				// Fetch the output value of 'table_name' from the state
+				rs, ok := s.RootModule().Outputs["table_name"]
+				if !ok || rs.Value == "" {
+					return "", fmt.Errorf("table_name output not found in the first step")
+				}
+				return rs.Value.(string), nil
+			},
```

Comment:
We should probably make a helper function for this, as this will be a very common case for testing import:
```go
func RetrieveAttributeFromResourceFunc(resourceId, attr string) func(*terraform.State) (string, error) {
  return func(s *terraform.State) (string, error) {
    rs, ok := s.RootModule().Resources[resourceId]
    if !ok {
      return "", fmt.Errorf("resource %s does not exist", resourceId)
    }
    id, ok := rs.Primary.Attributes[attr]
    if !ok || id.Value == "" {
      return "", fmt.Errorf("attribute %s not defined on resource %s", attr, resourceId)
    }
    return id, nil
  }
}
```

Reply:
Great idea, updated